File: android/src/androidTest/java/org/ligi/passandroid/TheQuirkCorrector.java
package org.ligi.passandroid;

import android.test.suitebuilder.annotation.SmallTest;
import org.ligi.passandroid.model.pass.Pass;
import static org.assertj.core.api.Assertions.assertThat;

public class TheQuirkCorrector extends TheAppleStyleBarcodeReaderBase {

    @SmallTest
    public void testWestbahnDescriptionIsFixed() throws Exception {
        loadPassFromAsset("passes/workarounds/westbahn/special.pkpass", new OnPassLoadCallback() {
            @Override
            public void onPassLoad(Pass pass) {
                assertThat(pass.getDescription()).isEqualTo("Wien Westbahnhof->Amstetten");
            }
        });

    }

}

File: android/src/androidTest/java/org/ligi/passandroid/TheQuirkCorrector.kt
package org.ligi.passandroid

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.ligi.passandroid.functions.loadPassFromAsset

class TheQuirkCorrector {

    @Test
    fun testWestbahnDescriptionIsFixed() {
        loadPassFromAsset("passes/workarounds/westbahn/special.pkpass") {
            assertThat(it!!.description).isEqualTo("Wien Westbahnhof->Amstetten")
        }
    }

}
